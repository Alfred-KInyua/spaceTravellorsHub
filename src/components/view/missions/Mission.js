/* eslint-disable react/jsx-key */
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  fetchMissions, selectMissions, joinMission, leaveMission,
} from '../../../redux/Missions';
import 'bootstrap/dist/css/bootstrap.css';

const Mission = () => {
  const missions = useSelector(selectMissions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissions({ limit: 3 }));
  }, [dispatch]);

  return (
    <>
      <table className="table table-hover">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <td>{mission.mission_name}</td>
              <td>{mission.description}</td>
              {mission.reserved ? (
                <td>
                  <span className="badge text-bg-info p-3">Active Member</span>
                </td>
              ) : (
                <td>
                  <span className="badge text-bg-secondary p-3">Not A Member</span>
                </td>
              )}

              {mission.reserved ? (
                <td>
                  <button type="button" className="btn btn-outline-danger" onClick={() => dispatch(leaveMission(mission.mission_id))}>Leave Mission</button>
                </td>
              ) : (
                <td>
                  <button type="button" className="btn btn-outline-secondary" onClick={() => dispatch(joinMission(mission.mission_id))}>Join Mission</button>
                </td>
              )}
            </tr>
          ))}
        </tbody>
      </table>
    </>
  );
};

export default Mission;
