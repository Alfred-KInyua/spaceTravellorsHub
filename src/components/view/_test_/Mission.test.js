import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../../../app/Store';
import Mission from '../missions/Mission';

const MockMission = () => (
  <Provider store={store}>
    <Mission />
  </Provider>
);

it('should match snapshot of the mission component', () => {
  const tree = renderer.create(<MockMission />).toJSON();
  expect(tree).toMatchSnapshot();
});
