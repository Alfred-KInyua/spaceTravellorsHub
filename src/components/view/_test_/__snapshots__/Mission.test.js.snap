// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot of the mission component 1`] = `
<table
  className="table table-hover"
>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
